plugins/speculation-rules/settings.php

```diff
@@ -281,9 +309,39 @@ function plsr_render_settings_field( array $args ): void {
 		<?php endforeach; ?>
 
 		<p class="description" style="max-width: 800px;">
-			<?php echo esc_html( $args['description'] ); ?>
+			<?php
+			echo wp_kses(
+				$args['description'],
+				array(
+					'a' => array(
+						'href'   => array(),
+						'target' => array(),
+					),
+				)
+			);
+			?>
 		</p>
 	</fieldset>
+	
+	<?php if ( 'authentication' === $args['field'] && ! $has_object_cache ) : ?>
+		<script>
+		( function () {
+			const authRadios = document.querySelectorAll( '.plsr-auth-radio' );
+			const warningDiv = document.getElementById( 'plsr-auth-warning' );
+			const authenticatedOptions = <?php echo wp_json_encode( $authenticated_options ); ?>;
```

Comment:
To be extra safe:

```suggestion
			const authenticatedOptions = <?php echo wp_json_encode( $authenticated_options, JSON_HEX_TAG | JSON_UNESCAPED_SLASHES ); ?>;
```

See https://sirre.al/2025/08/06/safe-json-in-script-tags-how-not-to-break-a-site/

Related:
- https://github.com/WordPress/WordPress-Coding-Standards/issues/63
- https://github.com/WordPress/WordPress-Coding-Standards/issues/2575